### PR TITLE
Add Apache 2 license

### PR DESCRIPTION
**Pull request by dobermai**
Adds the Apache 2 license file to the project. Also updates all class files and adds IntelliJ meta-file for automatic Apache 2 license headers for all project files

### CODE REVIEW

File: .idea/copyright/Apache2.xml [hunk 1]
```diff
@@ -0,0 +1,6 @@
+<component name="CopyrightManager">
+  <copyright>
+    <option name="notice" value="&#10;/**&#10; * Copyright &amp;#36;today.year The MQTT Bee project.&#10; *&#10; * Licensed under the Apache License, Version 2.0 (the &quot;License&quot;);&#10; * you may not use this file except in compliance with the License.&#10; * You may obtain a copy of the License at&#10; *&#10; *     http://www.apache.org/licenses/LICENSE-2.0&#10; *&#10; * Unless required by applicable law or agreed to in writing, software&#10; * distributed under the License is distributed on an &quot;AS IS&quot; BASIS,&#10; * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.&#10; * See the License for the specific language governing permissions and&#10; * limitations under the License.&#10; */" />
```
Comment (SgtSilvio) [inline .idea/copyright/Apache2.xml]: Are the '*' necessary? It seems that IntelliJ generates them on its own.

File: src/main/java/org/mqttbee/annotations/DoNotImplement.java [hunk 2]
```diff
@@ -1,3 +1,22 @@
+/*
+ *
+ * *
+ *  * Copyright 2018 The MQTT Bee project.
```
Comment (SgtSilvio) [inline src/main/java/org/mqttbee/annotations/DoNotImplement.java]: '*' is part of the template and generated by IntelliJ.

Comment (dobermai) [reply src/main/java/org/mqttbee/annotations/DoNotImplement.java]: You're completely right. Please ignore the PR, I'll create a new one. Seems I completely missed that

**Comment by dobermai** [comment]
Closing PR as this PR has some problems

**Pull request closed without merging**